pkg/util/stringutils.go

```diff
@@ -3,25 +3,27 @@ package util
 import (
 	"crypto/sha512"
 	"regexp"
-	"strconv"
 	"strings"
 )
 
-var (
-	reTrue = regexp.MustCompile("(?i)^true|yes|1$")
-)
-
+// Sanitize ensures a string only contains alphanumeric characters, and starts
+// with a letter
 func Sanitize(str string) string {
 	str = strings.ToLower(str)
 	// replace all non-alphanumeric characters by "-"
 	str = regexp.MustCompile("[^a-z0-9-]").ReplaceAllString(str, "-")
 	// replace multiple subsequent "-" by a single "-"
 	str = regexp.MustCompile("-+").ReplaceAllString(str, "-")
 	// remove "-" and numbers at beginning, "-" at end of string
-	str = regexp.MustCompile("^[0-9-]*([^-]*)-?$").ReplaceAllString(str, "${1}")
+	str = regexp.MustCompile("^[0-9-]*(.*)-?$").ReplaceAllString(str, "${1}")
```

Comment:
This is still not correct because the "(.*)" will now also capture the trailing "-", hence trailing "-" don't get removed.

Reply:
Thanks, fixed & added test cases for it